### PR TITLE
chore(): refactor admob integration code Part 1

### PR DESCRIPTION
**Pull request by gunhansancar**
Refactor admob integration code Part 1 which includes converting the BannerAd to Swift.
The second part will contain the same for interstitial and video ads

Also this PR fixes compile issue when we upgrade the dependency of `SuperAwesome` sdk from 7.x to 8.5

## JIRA
[AAG-2724]

## DESCRIPTION
Refactor admob integration code to support newer versions of `AwesomeAds`

[AAG-2724]: https://superawesomeltd.atlassian.net/browse/AAG-2724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: SAAdMobAdapter.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    
+    static func adapterVersion() -> GADVersionNumber {
+        let versionNumber = AwesomeAds.info()?.versionNumber ?? ""
+        let versionComponents = versionNumber.components(separatedBy: ".")
+        var version = GADVersionNumber()
+        if versionComponents.count == 3 {
+          version.majorVersion = Int(versionComponents[0]) ?? 0
+          version.minorVersion = Int(versionComponents[1]) ?? 0
+            version.patchVersion = (Int(versionComponents[2]) ?? 0) * 100
```
Comment (me2k) [inline SAAdMobAdapter.swift:27]: Alignment here.

I see you also use 2 space indentation and not 4 as used else where? Maybe your XCode editor settings need a tweak?

File: SAAdMobAdapter.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        let versionNumber = AwesomeAds.info()?.versionNumber ?? ""
+        let versionComponents = versionNumber.components(separatedBy: ".")
+
+        if versionComponents.count >= 3 {
+          let majorVersion = Int(versionComponents[0]) ?? 0
+          let minorVersion = Int(versionComponents[1]) ?? 0
+          let patchVersion = Int(versionComponents[2]) ?? 0
+
+          return GADVersionNumber(
+            majorVersion: majorVersion, minorVersion: minorVersion, patchVersion: patchVersion)
```
Comment (me2k) [inline SAAdMobAdapter.swift:42]:
```suggestion
          return GADVersionNumber(majorVersion: majorVersion, 
                                  minorVersion: minorVersion,
                                  patchVersion: patchVersion)
```

File: SAAdMobAdapter.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        bannerAd?.loadBanner(for: adConfiguration, completionHandler: completionHandler)
+    }
+    
+    func loadInterstitial(for adConfiguration: GADMediationInterstitialAdConfiguration, completionHandler: @escaping GADMediationInterstitialLoadCompletionHandler) {
+        
+    }
+    
+    func loadRewardedAd(for adConfiguration: GADMediationRewardedAdConfiguration, completionHandler: @escaping GADMediationRewardedLoadCompletionHandler) {
+        
+    }
```
Comment (me2k) [inline SAAdMobAdapter.swift:63]: Empty methods, are these needed?

File: SAAdMobBannerAd.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+//
+
+import GoogleMobileAds
+import SuperAwesome
+
+@objc(SAAdMobBannerAd)
+class SAAdMobBannerAd: NSObject, GADMediationBannerAd {
+    var view: UIView {
+        return bannerAd ?? UIView()
+    }
```
Comment (me2k) [inline SAAdMobBannerAd.swift:15]:
```suggestion
class SAAdMobBannerAd: NSObject, GADMediationBannerAd {

    var view: UIView {
       bannerAd ?? UIView()
    }
```

File: SAAdMobBannerAd.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    
+    required override init() {
+        super.init()
+    }
```
Comment (me2k) [inline SAAdMobBannerAd.swift:19]: Is this really required? Surely the init in the super is fine?

**Review by thomas-james-1986 — APPROVED** [review]
Looks good!

File: SAAdMobBannerAd.swift [hunk 6]
```diff
@@ -31,14 +31,13 @@ class SAAdMobBannerAd: NSObject, GADMediationBannerAd {
...
-        }
+        let parameter = adConfiguration.credentials.settings["parameter"] as? String ?? ""
+        let placementId = Int(parameter) ?? 0
         let adSize = adConfiguration.adSize
-        print(adSize.size)
+        
         bannerAd = BannerView(frame: CGRect(x: 0, y: 0, width: adSize.size.width, height: adSize.size.height))
-        bannerAd?.setCallback({ placementId, event in
-            self.onEvent(placementId, event)
+        bannerAd?.setCallback({ [weak self] placementId, event in
```
Comment (thomas-james-1986) [inline SAAdMobBannerAd.swift:39]: 👍

File: SAAdMobBannerAd.swift [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    
+    func loadBanner(
+        for adConfiguration: GADMediationBannerAdConfiguration,
+        completionHandler: @escaping GADMediationBannerLoadCompletionHandler
+    ) {
+        let parameter = adConfiguration.credentials.settings["parameter"] as? String ?? ""
+        let placementId = Int(parameter) ?? 0
+        let adSize = adConfiguration.adSize
+        
+        bannerAd = BannerView(frame: CGRect(x: 0, y: 0, width: adSize.size.width, height: adSize.size.height))
```
Comment (me2k) [inline SAAdMobBannerAd.swift:38]:
```suggestion
        bannerAd = BannerView(frame: CGRect(x: 0.0, y: 0.0, width: adSize.size.width, height: adSize.size.height))
```

One of the things to fractionally improve compile times is to get in to the habit of declaring float/doubles with .0

File: SAAdMobBannerAd.swift [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    
+    private func adLoaded() {
+        if let handler = completionHandler {
+            delegate = handler(self, nil)
+        }
+        bannerAd?.play()
+    }
+    
+    private func adError(message: String) {
+        let error = NSError(domain: "tv.superawesome.SAAdMobBannerAd",
```
Comment (me2k) [inline SAAdMobBannerAd.swift:65]:
```suggestion
        let error = NSError(domain: domain,
```

I would move this domain somewhere else, I believe you can get hold of it from the bundle of the framework too.
Something like: 
```
let bundle = Bundle(for: SAAdMobBanner.self)
let bundleId = bundle.bundleIdentifier
```

**Review by me2k — CHANGES_REQUESTED** [review]
I've left a few comments, mostly around clean up. Great work 👏

File: Adapters/AdMob/Classes/SAAdMobBannerCustomEvent.m [hunk 9]
```diff
@@ -61,7 +59,7 @@ - (void) requestBannerAd:(GADAdSize)adSize
     // Step 4. Add callbacks
     [_bannerAd setCallback:^(NSInteger placementId, SAEvent event) {
```
[the remark this answers is not in the document]
Comment (gunhansancar) [reply Adapters/AdMob/Classes/SAAdMobBannerCustomEvent.m:60]: Yes, we can get rid of all of these obj-c classes next week when we have more time.

File: SAAdMobAdapter.swift (hunk 1 above)
Comment (gunhansancar) [reply SAAdMobAdapter.swift:27]: We use swiftlint actually but for some reason it doesn't fix this inner project :/ I'll manually fix it

File: SAAdMobBannerAd.swift (hunk 8 above)
Comment (gunhansancar) [reply SAAdMobBannerAd.swift:65]: We already have it in AwesomeAds.info.bundle I'll use that instead 🦖

File: SAAdMobBannerAd.swift (hunk 7 above)
Comment (gunhansancar) [reply SAAdMobBannerAd.swift:38]: Nice one

File: SAAdMobBannerAd.swift (hunk 5 above)
Comment (gunhansancar) [reply SAAdMobBannerAd.swift]: Removed

**Review by me2k — APPROVED** [review]

**Pull request merged**